core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1692,64 +1712,66 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
     }
   }
 
-  private[zio] final case class BitChunk(bytes: Chunk[Byte], minBitIndex: Int, maxBitIndex: Int)
-      extends Chunk[Boolean] {
+  private[zio] sealed abstract class BitChunkBase[T](
+    chunk: Chunk[T],
+    val x: Int,
```

Comment:
Isn't there really just one parameter here that is like `bits` and then `x` is `bits` and `x2` is `2^bits - 1`?

Reply:
I found a solution, I guess.